### ISSUE TITLE
False positive comparison-with-callable when comparing to Any in python 3.10

### ISSUE THREAD

**Issue by XUWeijiang** [PyCQA/pylint#5557]
### Bug description

When I run pylint on the following code (saved as `a.py`):
```python
from typing import Any

def check_any(type_) -> bool:
    return type_ == Any
```
I get an warning:
```
a.py:4:11: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:3:0: C0116: Missing function or method docstring (missing-function-docstring)
a.py:4:11: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
```

### Expected behavior

No `comparison-with-callable` warning.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.0 | packaged by conda-forge | (default, Nov 10 2021, 13:20:59) [MSC v.1916 64 bit (AMD64)]
```

### OS / Environment

WIndows 10 (FYI: I can also reproduce this bug in Ubuntu 20.04)

**Comment by jacobtylerwalls** [comment]
Reproduced on Python 3.9.6
